### ISSUE TITLE
Note really an issue but I am trying to understand

### ISSUE THREAD

**Issue by harfordhawk** [hobbzey/MQTT_SmartBlinds#1]
Would like to contact you about this script.  Trying to understand how to adjust how much rotation is commanded for open and close.  thanks

**Comment by hobbzey** [comment]
No prob bob! Line 145 https://github.com/hobbzey/MQTT_SmartBlinds/blob/master/NodeMCU_MQTT_Sketch.ino#L145 set the amount of degrees to goto to open (right now it's at 90). 
The code says `for (pos = 0; pos <= 90; pos += 1) `
so basically it's saying move from 0 degrees to 90 degrees, 1 degree at a time

and line 156 https://github.com/hobbzey/MQTT_SmartBlinds/blob/master/NodeMCU_MQTT_Sketch.ino#L156 sets the degrees to goto to close (right now it goes back to 0).
The code says `for (pos = 90; pos >= 0; pos -= 1) `
so basically it's saying move back from 90 degrees to 0 degrees, 1 degree at a time

Does that make sense?

**Comment by harfordhawk** [comment]
yes, makes sense, but I want to rotate a servo probably 5-7 rotations? do
it just assign it 360 X number of rotations?  thanks

**Comment by hobbzey** [comment]
Oh! This was designed with taking out the gearbox of the blinds and having the servo coupled directly to the metal axle where the gearbox was with one of these critters: http://www.amazon.com/6mm-Hitec-Servo-Shaft-Coupler/dp/B00P2LYSB4 and this is why I only called for 90 degrees of rotation. And this installation also proved to be very clean with the board and servo completely concealed in the blinds upper housing!

Are you perhaps trying to leave the gear box in-place and just rotate the plastic rotation "wand" hanging down from it?

**Comment by harfordhawk** [comment]
yes, exactly.  I was using Ben's script ( that is where I found the link to yours), and his calls for a spin time, which should work great, but for some reason, I can get only one spin , then it just stops working, until I disconnect, wait, etc I am not really sure why it works sometimes and then refuses to work..

**Comment by hobbzey** [comment]
Ahhhhh! First things first, Ben is using a continuous rotation servo, if you want multiple rotations this is a must because standard servos can't go past 180 or 360 degrees (depending on model) but if you do in-fact have a continuous rotation servo then yeah the spintime method should work if you replace lines 139 through 166 with this code:

```
void servo_move() {
  Serial.println("State Change. Rotating Servo");
  if ( dirc == 180) {
    blindstatuspub.publish("close");                // change the MQTT state feed to close
	myservo.attach(servoPin);                   // energize servo
        myservo.write(dirc);
        delay(5000);                                // spin for 5 seconds
	myservo.detach();                           // movement finished so detach servo to conserve power
  }
  else if (dirc == 0) {
    blindstatuspub.publish("open");                 // change the MQTT state feed to open
	myservo.attach(servoPin);                   // energize servo
        myservo.write(dirc);
        delay(5000);                                // spin for 5 seconds
	myservo.detach();                           // movement finished so detach servo to conserve power
  }

  Serial.println("Returning to main loop");
  return;
}

```

**Comment by harfordhawk** [comment]
yeah, thanks for that.  I was trying your code as I could not get ben's to work properly. It will just randomly accept a command and work, but for the most part it just does nothing when I push either open or close.  I have tried 2 mqtt servers and finally set up my own, but all with the same results.. I am trying to figure out what is happening.  thanks for your time

**Comment by hobbzey** [comment]
Again, no problem! I actually dove into all this with the MQTT just to get my blinds, and any future ESP8266 projects, to work with my SmartThings hub that I'm already heavily invested in.

But in the end I found the MQTT route to be overly complicated and unreliable and actually discovered a way to simplify the process from running a MQTT bridge and also a MQTT server; down to just smartthings to ESP8266 directly and cut out the MQTT entirely!

Soooo, if your end goal like mine is to just control your blinds with SmartThings I actually have an even simpler solution I can send you when I get home...

**Comment by harfordhawk** [comment]
that's great, thanks for your help again... well I am using homeassistant.
I have a wink hub, but thats really it. I might be stuck with mqtt with
homeassistant.
